onnxruntime/test/providers/qnn/qnn_test_utils.h

```diff
@@ -78,12 +78,57 @@ struct TestInputDef {
     return std::get<RawData>(data_info_).data;
   }
 
+  std::pair<T, T> GetRange() const {
+    auto which_type = data_info_.index();
+    std::pair<T, T> range;
+
+    if (which_type == 0) {
+      // Get min/max of raw data.
+      range.first = std::numeric_limits<T>::min();
+      range.second = std::numeric_limits<T>::max();
+
+      for (auto val : std::get<RawData>(data_info_).data) {
+        range.first = std::min(range.first, val);
+        range.second = std::max(range.second, val);
+      }
+    } else {
+      assert(which_type == 1);
+      RandomData rand_info = std::get<RandomData>(data_info_);
+      range.first = rand_info.min;
+      range.second = rand_info.max;
+    }
+
+    return range;
+  }
+
  private:
   std::vector<int64_t> shape_;
   std::variant<RawData, RandomData> data_info_;
   bool is_initializer_;
 };
 
+template <typename QType = uint8_t>
+struct QuantParams {
+  float scale;
+  QType zero_point;
+
+  static QuantParams<QType> Compute(float rmin, float rmax) {
+    const QType qmin = std::numeric_limits<QType>::min();
+    const QType qmax = std::numeric_limits<QType>::max();
+
+    const float scale = (rmax - rmin) / static_cast<float>(qmax - qmin);
+    const QType zero_point = (static_cast<float>(qmin) - rmin) / scale;
+
+    return QuantParams<QType>{scale, zero_point};
+  }
+};
+
+template <typename QType = uint8_t>
+inline QuantParams<QType> GetTestInputQuantParams(const TestInputDef<float>& input_def) {
```

Comment:
We have a lot of qdq unit tests that use seemingly random quantization params. This util allows us to compute the appropriate quantization params for a given test input.

To get the quant params for the output, we would instead pass the expected output range to the `QuantParams::Compute()` function. I'll go through tests and update as necessary in a separate PR.